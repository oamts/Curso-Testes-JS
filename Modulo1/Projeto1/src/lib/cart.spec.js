import Cart from './cart';

describe('Cart', () => {
  let cart;
  let product = {
    title: 'shoes',
    price: '35388', //355.88 | R$383,88
  };

  beforeEach(() => {
    cart = new Cart();
  });

  it('should return 0 when getTotal is executed in a newly created instance', () => {
    expect(cart.getTotal()).toEqual(0);
  });

  it('should multiply quantity and price and recieve the total amount', () => {
    const item = {
      product,
      quantity: 2,
    };

    cart.add(item);

    expect(cart.getTotal()).toEqual(70776);
  });

  fit('should ensure no more than one procuct exists at a time', () => {
    cart.add({
      product,
      quantity: 2,
    });

    cart.add({
      product,
      quantity: 1,
    });

    expect(cart.getTotal()).toEqual(35388);
  });
});
