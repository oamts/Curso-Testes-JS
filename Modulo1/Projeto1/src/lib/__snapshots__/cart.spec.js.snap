// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cehckout() should return an object with the total and the list of items 1`] = `
{
  "items": [
    {
      "product": {
        "price": "35388",
        "title": "shoes - men",
      },
      "quantity": 2,
    },
    {
      "product": {
        "price": "41872",
        "title": "shoes - women",
      },
      "quantity": 3,
    },
  ],
  "total": 196392,
}
`;
